jit/analyze.ts

```diff
@@ -76,14 +79,44 @@ module J2ME {
     "gnu/testlet/vm/NativeTest.returnAfterPause.()I": YieldReason.Root,
     "gnu/testlet/vm/NativeTest.dumbPipe.()Z": YieldReason.Root,
     "gnu/testlet/TestHarness.getNumDifferingPixels.(Ljava/lang/String;)I": YieldReason.Root,
+
+
   };
 
-  export function isFinal(classInfo: ClassInfo): boolean {
-    if (classInfo.isFinal) {
-      return true;
+  export var yieldVirtualMap = {
+    // These can technically yield but are worth the risk.
+    // "java/lang/Object.equals.(Ljava/lang/Object;)Z": YieldReason.None
+  }
+
+  export function isFinalClass(classInfo: ClassInfo): boolean {
+    var result = classInfo.isFinal;
+    if (!result) {
+      result = classInfo.subClasses.length === 0;
     }
+    // console.log(classInfo.className + " is final class " + result);
+    return result;
+  }
+
+  export function isFinalMethod(methodInfo: MethodInfo): boolean {
     return false;
-    // TODO: Be more clever here.
+    var result = methodInfo.isFinal;
```

Comment:
Erm, are we now able to be more clever here? We're still returning false unconditionally.